subsys/zigbee/lib/zigbee_app_utils/zigbee_app_utils.c

```diff
@@ -856,3 +868,47 @@ void zigbee_configure_sleepy_behavior(bool enable)
 	}
 }
 #endif
+
+/* Wrapper allowing to schedule zb_bdb_reset_via_local_action as an alarm */
+static void factory_reset(zb_uint8_t arg)
+{
+	ZVUNUSED(arg);
+	LOG_DBG("Perform factory reset");
+	zb_bdb_reset_via_local_action(0);
+}
+
+static void factory_reset_callback(struct k_timer *timer_id)
+{
+	uint32_t button_state = 0;
+	uint32_t has_changed = 0;
+
+	dk_read_buttons(&button_state, &has_changed);
+	if (button_state & factory_reset_context.button) {
+		LOG_DBG("FR button pressed for %d [s]", timer_id->status);
+		if (timer_id->status >= factory_reset_context.press_time) {
+			ZB_SCHEDULE_APP_ALARM(factory_reset, 0, FACTORY_RESET_DELAY);
+			LOG_DBG("Schedule factory reset");
+			k_timer_stop(timer_id);
+		}
+	} else   {
+		LOG_DBG("FR button released prematurely");
+		k_timer_stop(timer_id);
+	}
+}
+
+void register_factory_reset_button(uint32_t button, uint32_t press_time_seconds)
```

Comment:
Why do we need to pass the `press_time_seconds` if the value is configured through the Kconfig?

Reply:
@tomchy Not needed anymore after the configuration approach has changed. Will remove the unnecessary parameter.